building-with-ocean/build-a-marketplace/running-isolated-market.md

```diff
@@ -0,0 +1,135 @@
+# Running Isolated Market
+
+An isolated market can determine which assets can be listed in marketplace interface. An isolated market requires an **Aquarius** instance that serves the asset metadata. When an asset is published using a such Aquarius instance, it checks if the validator belongs to the list of allowed validators. If the metadata proof is valid and signed by a permitted validator, Aquarius caches the asset metadata. After caching, marketplace can fetch the asset details and show it on the marketplace UI.
+
+### Use case
+
+Suppose Alice and Bob are running their own Aquarius instance and caching assets independently. Now, using isolated an isolated market, Charlie can run a market place that lists assets that are allowed to be stored in both Alice and Bob's Aquarius instances. Here, Charlie would allow wallet addresses of Alice and Bob's Aquarius instances in `ALLOWED_VALIDATORS`. Thus, it is possible to compose marketplaces that list the same asset without the need to republish for each market individually.&#x20;
+
+### Deploy Aquarius
+
+The details on deploying Aquarius are on [this page](../deploying-components/deploying-aquarius.md).  Add the `ALLOWED_VALIDATORS` in the [`.env`](../deploying-components/deploying-aquarius.md#create-a-.env-file) file. `ALLOWED_VALIDATORS` is a JSON array of allowed addresses that can act as validators. A validator has to sign the hash of the metadata asset as proof which is then validated by Aquarius.&#x20;
+
+`ALLOWED_VALIDATORS` is a list of addresses. e.g., `["0xBE5449a6A97aD46c8558A3356267Ee5D2731ab5e","`0xe2DD09d719Da89e5a3D0F2549c7E24566e947260`"]`
+
+If the asset includes metadata proofs generated be any of the addresses in `ALLOWED_VALIDATORS`, then Aquarius will cache the asset.&#x20;
+
+### Deploy Market
+
+The details on deploying Market are on [this page](../deploying-components/deploying-marketplace.md). The guide on [customizing market](customising-your-market.md) gives details about changing the theme, colors, font that suits the your requirements. Set the `NEXT_PUBLIC_METADATACACHE_URI` variable in the marketplace [.env file](../deploying-components/deploying-marketplace.md#create-file-with-name-.env) to the correct Aquarius URL hosted in the [previous step](running-isolated-market.md#deploy-aquarius).&#x20;
+
+### Add metadata proof to an asset with
+
+#### Generating metadata proof
```

Comment:
We need a lot more explanation here on how the user needs to adapt the market. The idea with this guide is that it follows on from the series of how to create a marketplace. The user has forked the market repo, they've customised and deployed it and now the next step is to make their market isolated. As a user I would want to know the following:

- What files in the market need to be edited? 
- What envs are required and where do they need to be saved? 
- Is this code snippet supposed to be saved as a file? If so, what should it be called and where should it be saved?
- All of the other guides in this series are just in js so a user will be wondering why they have been given python code here and what to do with it. For a beginner that will be confusing... 
- As a general rule, lets always include the commands to run the code once it is saved

Reply:
It would also be really nice to include screenshots of you editing the market code, just like we did in the [customisation guide](https://docs.oceanprotocol.com/building-with-ocean/build-a-marketplace/customising-your-market)

Reply:
Regarding: 
What files in the market need to be edited?
What envs are required and where do they need to be saved?
and screenshots:

We already have dedicated page on deploying a market and customizing market. So, the current page is already referring to those and only providing additional steps. Thus, we are not repeating the steps. Do we want to repeat the steps to clone, customize the market again here?

Is this code snippet supposed to be saved as a file? If so, what should it be called and where should it be saved?
 
-> Agreed, I shall add more context on the purpose of the code snippets

Reply:
Nothing should be repeated from the other guides, this should pick up from where they left off. The point I was trying to make is that we need to cover all of the files that need to be edited or any envs that need to be added (if there are none, no need to mention it). If existing code from the market needs to be edited it would be nice to have screenshots